pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -372,7 +372,7 @@ String getHandlerName() {
     }
 
     // This listener is triggered when topics partitions are updated.
-    private class TopicsPartitionChangedListener implements PartitionsChangedListener {
+    public class TopicsPartitionChangedListener implements PartitionsChangedListener {
```

Comment:
Better not change this to `public`.
Using PartitionedProducerImpl#`getPartitionsAutoUpdateFuture` could help the test